arch/xtensa/core/CMakeLists.txt

```diff
@@ -22,6 +22,10 @@ zephyr_library_sources_ifdef(CONFIG_DEBUG_COREDUMP coredump.c)
 
 zephyr_library_sources_ifdef(CONFIG_GDBSTUB gdbstub.c)
 
+if("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "xcc")
```

Comment:
Will need to think about how to proceed.

Though, for XCC, the missing `atexit()` is probably due to mis-configuration when the toolchain was built. It is only missing for one SoC-specific XCC. Others build fine. Maybe we can have a "toolchain supplement" or overlay instead of scattering all toolchain-specific bits all over the tree.

Reply:
yes, I was not requesting any changes atm.
Just pointing out my observations.